src/wallet/rpcwallet.cpp

```diff
@@ -3222,8 +3222,6 @@ Value z_sendmany(const Array& params, bool fHelp)
             throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, expected object");
         const Object& o = output.get_obj();
 
-        RPCTypeCheck(o, boost::assign::map_list_of("address", str_type)("amount", real_type));
```

Comment:
sendtoaddress just uses `AmountFromValue` which we use later too.  The never use RPCTypeCheck.  sendmany doesn't use RPCTypeCheck either.  I added this check as part of z_sendmany, but it is too strict.